# STAT 207 Group Lab Assignment 11 - [10 total points]

## Logistic Regression

<hr>

## <u>Lab Grading</u>:

Should we grade your submission?  If not, write the netID of the submission to be graded.  (Note: We will only grade one assignment per group, and we'll pick the first one that says we should grade that submission.  We will assign the same grade to all team members.)

*For example*, you might respond: **grade this submission** or **my submission is under netID jdeeke**

If you said **my submission is under netID** above, we will not read any more of your lab submission.

If you said **grade this submission** above, who worked with you on this submission?  Write both their **names** and **netIDs**.  

Also, discuss and record if you follow the NBA + your favorite NBA team, if you have one.

<hr>

## <u>Purpose</u>:
You should work in groups of 2-3 on this report (not working in groups without permission will result in a point deduction). The purpose of this group lab assignment is to fit and use a logistic regression model.
<hr>

## <u>Assignment Instructions</u>:

### Group Roles

Suggested and specified roles are provided below: 

#### Groups of 2

* **Driver**: This student will type the report.  While typing the report, you may be the one who is selecting the functions to apply to the data.
* **Navigator**: This student will guide the process of answering the question.  Specific ways to help may include: outlining the general steps needed to solve a question (providing the overview), locating examples within the course notes, and reviewing each line of code as it is typed.

#### Groups of 3

* **Driver**: This student will type the report.  They may also be the one to select the functions to apply to the data.
* **Navigator**: This student will guide the process of answering the question.  They may select the general approach to answering the question and/or a few steps to be completed along the way. 
* **Communicator**: This student will review the report (as it is typed) to ensure that it is clear and concise.  This student may also locate relevant examples within the course notes that may help complete the assignment.

<hr>

### Imports

In [1]:
#Run this
import pandas as pd                    # imports pandas and calls the imported version 'pd'
import matplotlib.pyplot as plt        # imports the package and calls it 'plt'
import seaborn as sns                  # imports the seaborn package with the imported name 'sns'
sns.set()  

## Case Study: Understanding NBA Longevity

We will look at data collected on rookies in the NBA from 1980-2016 to determine if the rookies are successful and still playing in the NBA after 5 years.  The available variables include the following **per game** statistics:

- **GP**: number of games played
- **MIN**: number of minutes played per game
- **PTS**: number of points scored per game
- **FGM**: number of field goals made per game
- **FGA**: number of field goals attempted per game
- **FG%**: percentage of field goals made per game
- **3P Made**: number of 3-point field goals made per game
- **3PA**: number of 3-point field goals attempted per game
- **3P%**: percentage of 3-point field goals made per game
- **FTM**: number of free throws made per game
- **FTA**: number of free throws attempted per game
- **FT%**: percentage of free throws made per game
- **OREB**: number of offensive rebounds per game
- **DREB**: number of defensive rebounds per game
- **REB**: total number of rebounds per game
- **AST**: number of assists per game
- **STL**: number of steals per game
- **BLK**: number of blocks per game
- **TOV**: number of turnovers per game
- **TARGET_5Yrs**: whether the player is still playing in the NBA after 5 years (1 = playing, 0 = not playing)

The code cell below will read in the data for you.  Be sure to run the cell. 

In [3]:
df = pd.read_csv('nba_rookie.csv')
df

Name  GP   MIN  PTS  FGM  FGA  FG_Perc  3P Made  3PA  \
0      Brandon Ingram  36  27.4  7.4  2.6  7.6     34.7      0.5  2.1   
1     Andrew Harrison  35  26.9  7.2  2.0  6.7     29.6      0.7  2.8   
2      JaKarr Sampson  74  15.3  5.2  2.0  4.7     42.2      0.4  1.7   
3         Malik Sealy  58  11.6  5.7  2.3  5.5     42.6      0.1  0.5   
4         Matt Geiger  48  11.5  4.5  1.6  3.0     52.4      0.0  0.1   
...               ...  ..   ...  ...  ...  ...      ...      ...  ...   
1304      Chris Smith  80  15.8  4.3  1.6  3.6     43.3      0.0  0.2   
1305      Brent Price  68  12.6  3.9  1.5  4.1     35.8      0.1  0.7   
1306     Marlon Maxey  43  12.1  5.4  2.2  3.9     55.0      0.0  0.0   
1307  Litterial Green  52  12.0  4.5  1.7  3.8     43.9      0.0  0.2   
1308        Jon Barry  47  11.7  4.4  1.6  4.4     36.9      0.4  1.3   

      3P_Perc  ...  FTA  FT_Perc  OREB  DREB  REB  AST  STL  BLK  TOV  \
0        25.0  ...  2.3     69.9   0.7   3.4  4.1  1.9  0.4  0.4  1.3   
1        23.5  ...  3.4     76.5   0.5   2.0  2.4  3.7  1.1  0.5  1.6   
2        24.4  ...  1.3     67.0   0.5   1.7  2.2  1.0  0.5  0.3  1.0   
3        22.6  ...  1.3     68.9   1.0   0.9  1.9  0.8  0.6  0.1  1.0   
4         0.0  ...  1.9     67.4   1.0   1.5  2.5  0.3  0.3  0.4  0.8   
...       ...  ...  ...      ...   ...   ...  ...  ...  ...  ...  ...   
1304     14.3  ...  1.5     79.2   0.4   0.8  1.2  2.5  0.6  0.2  0.8   
1305     16.7  ...  1.0     79.4   0.4   1.1  1.5  2.3  0.8  0.0  1.3   
1306      0.0  ...  1.6     64.3   1.5   2.3  3.8  0.3  0.3  0.4  0.9   
1307     10.0  ...  1.8     62.5   0.2   0.4  0.7  2.2  0.4  0.1  0.8   
1308     33.3  ...  1.0     67.3   0.2   0.7  0.9  1.4  0.7  0.1  0.9   

      TARGET_5Yrs  
0               0  
1               0  
2               0  
3               1  
4               1  
...           ...  
1304            0  
1305            1  
1306            0  
1307            1  
1308            1  

[1309 rows x 21 columns]

### 1. [5.5 points] Fit a Logistic Regression Model

**a)** Fit a model predicting whether the player lasts 5 years or more in the NBA from four of the other variables in the data.  You may pick the combination of four variables to use as predictors.

**b)** What is the pseudo-$R^2$ for this model?  

**c)** In general, what is the pseudo-$R^2$ measuring?  In other words, what quality of a model is measured with this number?

**d)** Interpret the coefficient fit for one of the predictor variables in terms of the **odds**.

**e)** What are the baseline odds for the fitted model?

### 2. [2 points] Ayo Donsunmu

A recent player in the NBA from the University of Illinois is Ayo Donsunmu.  

**a)** Use the NBA Regular Season Stats per game for the 2021-2022 season (Ayo's rookie season) from this website https://basketball.realgm.com/player/Ayo-Dosunmu/Summary/107970 to predict the probability that Ayo will last in the NBA for at least 5 years.  Be sure to scroll down and use his statistics reported for the 2021-2022 season, corresponding to his rookie year.

**b)** Based on the available observations in the dataset, does this prediction seem trustworthy?  Briefly explain.

*Hint*: there are two ways that you can answer this question, both of which compare Ayo's characteristics to those in the available data.  One approach requires analysis of the data, while the other does not.

### 3. [2.5 points] Appropriate Model?

**a)** Is a logistic regression model the appropriate model for the data?  Or, should we use linear regression instead?  Briefly explain.

**b)** Generate and interpret a plot to assess the linearity assumption for this model.  Does this model seem to be an appropriate model for the data?